nemo_curator/core/utils.py

```diff
@@ -68,6 +71,47 @@ def _logger_custom_deserializer(
     return logger
 
 
+@tenacity.retry(
+    wait=tenacity.wait_fixed(1),
+    stop=tenacity.stop_after_delay(RAY_CLUSTER_START_VERIFICATION_TIMEOUT),
+    retry=tenacity.retry_if_result(lambda x: x is False),
+    reraise=True,
+)
```

Comment:
This is essentially operating like a `while True` and sleep right? Could we use it in this test https://github.com/NVIDIA-NeMo/Curator/blob/main/tests/core/test_get_ray_client.py#L84 ?

Reply:
Probably yes. This is not a blocker, though, right? I can address in my followup PR to this. The multinode integration, since that will touch the tests a lot.